src/mqtt_client/mod.rs

```diff
@@ -442,20 +446,10 @@ where
         F: FnMut(&str, &mut Settings, &Settings) -> Result<(), E>,
         E: AsRef<str>,
     {
-        let Self {
```

Comment:
Huh, I wonder what changed that made this no longer necessary. I'm glad we can get rid of it in any case!

Reply:
Yeah. This became unnecessary a while ago. It now accurately captures individual items in structs (and not the entire struct). Also https://doc.rust-lang.org/1.33.0/book/ch18-03-pattern-syntax.html#legacy-patterns-ref-and-ref-mut